app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
```

Comment:
Sorry for the confusion. The current behavior is not what I'm saying is wrong. What I'm saying is wrong is determining whether or not a context is LOA3 based on the user having a recovery code. Here is the problem, restated:

Steps to reproduce:

1. Make LOA3 request from SP
2. Click "Create account"
3. Enter email and confirm it
4. Enter phone and confirm it => You should be on `/verify` endpoint now
5. Start the IdV flow and fill out the personal information form and submit it => You should be on the financials info page now
6. Click on the "Cancel" link

Expected Result: I am prompted to Continue Id Verification or Cancel and return to Profile

Actual Result: I am prompted to Continue Account Creation or Cancel and delete your information.